src/main/java/stsjorbsmod/actions/ChainLightningAction.java

```diff
@@ -32,27 +32,18 @@ public ChainLightningAction(AbstractCreature owner, AbstractMonster initialTarge
     }
 
     public void update() {
-        ArrayList<AbstractMonster> remainingTargets = new ArrayList<>(allTargets);
-        int nextTargetIndex = remainingTargets.indexOf(initialTarget);
-        int currentDamage = this.damage;
         int multiplier = 0;
-
-        while(true) {
-            AbstractMonster nextTarget = remainingTargets.remove(nextTargetIndex);
-
-            if (!nextTarget.halfDead && !nextTarget.isDying && !nextTarget.isEscaping) {
-                AbstractDungeon.actionManager.addToBottom(
-                        new DamageAction(nextTarget, new DamageInfo(owner, currentDamage, DamageInfo.DamageType.NORMAL), this.attackEffect));
-                addLightningEffect(nextTarget, multiplier);
-                multiplier += 1;
-                currentDamage += this.extraDamagePerHop;
-            }
-
-            if (remainingTargets.isEmpty()) {
-                break;
-            }
-
-            nextTargetIndex = AbstractDungeon.cardRandomRng.random(0, remainingTargets.size() - 1);
+        int currentDamage = 0;
+        AbstractMonster currentTarget;
+        for (int i = 0; i < allTargets.size(); i++) {
+            currentTarget = allTargets.get(i);
+            currentDamage = this.damage[i];
+            if (currentDamage <= 0) continue;
+
+            AbstractDungeon.actionManager.addToBottom(

```

Comment:
I think at this point it's probably not necessary to have a separate action at all; I think we could probably do like ThunderClap does and just queue the `DamageAction`s and `VFXAction`s from the card directly and get rid of the intermediate damage/order arrays

Reply:
Yeah! That simplifies things a bit, for sure.